homeassistant/components/auth/__init__.py

```diff
@@ -598,6 +601,50 @@ async def websocket_delete_refresh_token(
     connection.send_result(msg["id"], {})
 
 
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "auth/delete_all_refresh_tokens",
+    }
+)
+@websocket_api.ws_require_user()
+@websocket_api.async_response
+async def websocket_delete_all_refresh_tokens(
+    hass: HomeAssistant, connection: websocket_api.ActiveConnection, msg: dict[str, Any]
+) -> None:
+    """Handle delete all refresh tokens request."""
+
+    async def remove_safely(token: RefreshToken) -> bool:
+        # For safety, a revoke_callback could raise an error.
+        try:
+            await hass.auth.async_remove_refresh_token(token)
+            return True
+        except Exception:  # pylint: disable=broad-except
+            return False
+
+    tasks = []
+    current_refresh_token: RefreshToken
+    for token in connection.user.refresh_tokens.values():
+        if token.id == connection.refresh_token_id:
+            current_refresh_token = token
+            continue  # Will be remove after sending the result
+        tasks.append(hass.async_create_task(remove_safely(token)))
+
+    remove_failed = False
+    if tasks:
+        for result in await asyncio.gather(*tasks):
+            if not result:
+                remove_failed = True
+
+    if remove_failed:
+        connection.send_error(
+            msg["id"], "token_removing_error", "During removal, an error was raised."
+        )
+    else:
+        connection.send_result(msg["id"], {})
+
+    hass.async_create_task(remove_safely(current_refresh_token))
```

Comment:
Could and should something track this task?

Reply:
Yeah, sure, but is that enough?

Reply:
We have a default exception handler for the loop set by our loop policy. It's still nice to handle exceptions explicitly in tasks, but if we don't expect any specific exceptions I'm not sure how much it will help.

Reply:
I guess the benefit would be that we could apply the same log message as we do for the other exceptions above.

Reply:
Should I create the `remove_safely` wrapper with the log entry?